### ISSUE TITLE
Build error related to expo-dev-menu when jsEngine=jsc

### ISSUE THREAD

**Issue by bndkt** [expo/expo#21233]
### Summary

I upgraded to Expo 48, which has Hermes enables by default and my build was successful. However, Hermes is leading to my iOS build being really big which is an issue because I'm building an App Clip which has a size limit of 15 MB. So I switched back to jsc (using the jsEngine param in app.json), which leads to the following build errors in EAS Build:

```
❌  (../../node_modules/expo-dev-menu/vendored/react-native-reanimated/ios/native/NativeProxy.mm:27:10)

  25 | #else
  26 | #if RNVERSION >= 71
> 27 | #include <jsc/JSCRuntime.h>
     |          ^ 'jsc/JSCRuntime.h' file not found
  28 | #else
  29 | #include <jsi/JSCRuntime.h>
  30 | #endif // RNVersion
```

and

```
▸ 	CompileC /Users/expo/Library/Developer/Xcode/DerivedData/Feld-ejrefokefgxjrfcfsvxvjoonuons/Build/Intermediates.noindex/ArchiveIntermediates/Feld/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/expo-dev-menu.build/Objects-normal/arm64/NativeProxy.o /Users/expo/workingdir/build/node_modules/expo-dev-menu/vendored/react-native-reanimated/ios/native/NativeProxy.mm normal arm64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'expo-dev-menu' from project 'Pods')
```

### Managed or bare workflow?

managed

### What platform(s) does this occur on?

iOS

### Package versions

"expo": "48.0.1",
"expo-dev-client": "~2.1.1",
"react-native-reanimated": "~2.14.4",

### Environment

expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.2.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 18.12.1 - ~/Library/pnpm/node
      Yarn: 1.22.19 - ~/Developer/GitHub/feld/node_modules/.bin/yarn
      npm: 8.19.2 - ~/Library/pnpm/npm
      Watchman: 2023.02.13.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    IDEs:
      Xcode: 14.2/14C18 - /usr/bin/xcodebuild
    Expo Workflow: managed

### Reproducible demo

- Create new Expo app
- Add expo-dev-client
- Add jsEngine="jsc" in app.json
- Build app

**Comment by brentvatne** [comment]
repro'd it here: https://github.com/brentvatne/jsc-dev-client-48

best to include a link to a github repo to avoid autoclosing in the future :)

**Comment by bndkt** [comment]
Thanks @brentvatne, will do! I found an issue that was fixed in react-native-reanimated@2.14.3: https://github.com/software-mansion/react-native-reanimated/issues/3965. Maybe upgrading the vendored version of reanimated helps?
